osutil/testhelper.go

```diff
@@ -38,3 +42,93 @@ func MustBeTestBinary(panicMsg string) {
 		panic(panicMsg)
 	}
 }
+
+var injectSysroot = "/"
+var stderr io.Writer = os.Stderr
+var foreverLoop = func() {
+	for {
+	}
+}
+
+func injectFault(tagKind string) (injected bool) {
+	s := strings.Split(tagKind, ":")
+	if len(s) != 2 {
+		fmt.Fprintf(stderr, "incorrect fault tag: %q\n", tagKind)
+		return false
+	}
+	tag := s[0]
+	kind := s[1]
+	stampFile := filepath.Join(injectSysroot, "/var/lib/snapd/faults", tagKind)
+	if FileExists(stampFile) {
+		// already injected once
+		return false
+	}
+
+	if err := os.MkdirAll(filepath.Join(injectSysroot, "/var/lib/snapd/faults"), 0755); err != nil {
+		fmt.Fprintf(stderr, "cannot create fault stamps directory: %v\n", err)
+		return false
+	}
+	makeStamp := func() bool {
+		if err := AtomicWriteFile(stampFile, nil, 0644, 0); err != nil {
+			fmt.Fprintf(stderr, "cannot create stamp file for tag %q\n", tagKind)
+			return false
+		}
+		return true
+	}
+	fmt.Fprintf(stderr, "injecting %q fault for tag %q\n", kind, tag)
+
+	switch kind {
```

Comment:
Not super important because there's a print just before but this will exit with true on an unsupported fault. It could have a `default` case that logs/returns an error. It'd just make the problem more obvious if I set the env var to an unsupported kind of fault by mistake, like SNAPD_FAULT_INJECT=tag:rebot